src/ModelContextProtocol/Server/McpServerExtensions.cs

```diff
@@ -210,4 +222,57 @@ async IAsyncEnumerable<ChatResponseUpdate> IChatClient.GetStreamingResponseAsync
         /// <inheritdoc/>
         void IDisposable.Dispose() { } // nop
     }
+
+    /// <summary>
+    /// Provides an <see cref="ILoggerProvider"/> implementation for creating loggers
+    /// that send logging message notifications to the client for logged messages.
+    /// </summary>
+    private sealed class ClientLoggerProvider(IMcpServer server) : ILoggerProvider
```

Comment:
It would be nice to expose this via an `AddClientLogger` extension method on ILoggingBuilder similar to [AddConsole](https://github.com/dotnet/runtime/blob/9559b2a8c2961fd72d2e687a2b3f8f148aa33eb5/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L38). I think that would be the easiest way to use this for hosted servers. We can do that as a follow up though.